sdk/servicebus/service-bus/test/streamingReceiver.spec.ts

```diff
@@ -974,6 +903,76 @@ describe("Streaming - User Error", function(): void {
   });
 });
 
+describe("Streaming - Failed init should not cache recevier", function(): void {
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  class TestTokenProvider extends SasTokenProvider {		
+    private firstCall = true;		
+    static errorMessage = "This is a faulty token provider.";
+    constructor(connectionObject: {
+      Endpoint: string,
+      SharedAccessKeyName: string,
+      SharedAccessKey: string
+    }) {		
+      super(		
+        connectionObject.Endpoint,		
+        connectionObject.SharedAccessKeyName,		
+        connectionObject.SharedAccessKey		
+      );		
+    }		
+
+     async getToken(audience: string): Promise<TokenInfo> {		
+      if (this.firstCall) {		
+        this.firstCall = false;		
+        throw new Error(TestTokenProvider.errorMessage);		
+      }		
+      return super.getToken(audience);		
+    }		
+  }
+
+  it("UnPartitioned Queue: Receiver is not cached when not initialized #RunInBrowser", async function(): Promise<
+    void
+  > {
+    const env = getEnvVars();
+
+    // Send a message using service bus client created with connection string
+    sbClient = getServiceBusClient();
+    let clients = await getSenderReceiverClients(sbClient, TestClientType.UnpartitionedQueue, TestClientType.UnpartitionedQueue);
+    sender = clients.senderClient.createSender();
+    await sender.send(TestMessage.getSample());
+    await sbClient.close();
+ 
+    // Receive using service bus client created with faulty token provider
+    const connectionObject: {
+      Endpoint: string,
+      SharedAccessKeyName: string,
+      SharedAccessKey: string
+    } = parseConnectionString(env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]);
+    const tokenProvider = new TestTokenProvider(connectionObject);
+    sbClient = ServiceBusClient.createFromTokenProvider(connectionObject.Endpoint.substr(5), tokenProvider);
+    clients = await getSenderReceiverClients(sbClient, TestClientType.UnpartitionedQueue, TestClientType.UnpartitionedQueue);
+    receiver = clients.receiverClient.createReceiver(ReceiveMode.peekLock);
+
+    let actualError: Error;
+    receiver.registerMessageHandler(async (msg: ServiceBusMessage) => {
+      throw new Error("No messages should have been received with faulty token provider");
+    }, (err) => {
+      actualError = err;
+    });
+
+    // Check for expected error and that receiver was not cached
+    const errCheck = await checkWithTimeout(() => !!actualError === true);
+    should.equal(errCheck, true, "Expected error to be thrown, but no error found.");
+    should.equal(actualError!.message, TestTokenProvider.errorMessage, 'Expected error from token provider, but unexpected error found.')
```

Comment:
Ok i commented on older PR earlier!
Was looking for this error message check - lgtm and is inline with decision to not retry and surfacing it to user